components/brave_wallet/browser/zcash/zcash_transaction_utils_unittest.cc

```diff
@@ -73,39 +110,190 @@ TEST(ZCashTransactionUtilsUnitTest, PickZCashTransparentInputs) {
     EXPECT_EQ(result->inputs[2].utxo_address, "30000");
     EXPECT_EQ(result->inputs[2].utxo_value, 30000u);
   }
+
+  // With change, Orchard output.
+  {
+    ZCashWalletService::UtxoMap utxo_map;
+    utxo_map["10000"] = GetZCashUtxo(10000);
+    utxo_map["20000"] = GetZCashUtxo(20000);
+    utxo_map["30000"] = GetZCashUtxo(30000);
+    utxo_map["40000"] = GetZCashUtxo(40000);
+
+    auto result = PickZCashTransparentInputs(utxo_map, 30000,
+                                             ZCashTargetOutputType::kOrchard);
+    EXPECT_TRUE(result);
+    EXPECT_EQ(result->change, 5000u);
+    // max(2, max(3, 1) + max(0, 1, 2)) * 5000.
+    EXPECT_EQ(result->fee, 25000u);
+    EXPECT_EQ(result->inputs[0].utxo_address, "10000");
+    EXPECT_EQ(result->inputs[0].utxo_value, 10000u);
+
+    EXPECT_EQ(result->inputs[1].utxo_address, "20000");
+    EXPECT_EQ(result->inputs[1].utxo_value, 20000u);
+
+    EXPECT_EQ(result->inputs[2].utxo_address, "30000");
+    EXPECT_EQ(result->inputs[2].utxo_value, 30000u);
+  }
+
+  // Max amount, Orchard output.
+  {
+    ZCashWalletService::UtxoMap utxo_map;
+    utxo_map["10000"] = GetZCashUtxo(10000);
+    utxo_map["20000"] = GetZCashUtxo(20000);
+    utxo_map["30000"] = GetZCashUtxo(30000);
+    auto result = PickZCashTransparentInputs(utxo_map, kZCashFullAmount,
+                                             ZCashTargetOutputType::kOrchard);
+    EXPECT_EQ(result->change, 0u);
+    // max(2, max(3, 0) + max(0, 1, 2)) * 5000.
+    EXPECT_EQ(result->fee, 25000u);
+    EXPECT_EQ(result->inputs[0].utxo_address, "10000");
+    EXPECT_EQ(result->inputs[0].utxo_value, 10000u);
+
+    EXPECT_EQ(result->inputs[1].utxo_address, "20000");
+    EXPECT_EQ(result->inputs[1].utxo_value, 20000u);
+
+    EXPECT_EQ(result->inputs[2].utxo_address, "30000");
+    EXPECT_EQ(result->inputs[2].utxo_value, 30000u);
+  }
+
+  // Not enough funds.
+  {
+    ZCashWalletService::UtxoMap utxo_map;
+    utxo_map["1"] = GetZCashUtxo(1);
+    utxo_map["2"] = GetZCashUtxo(2);
+    utxo_map["3"] = GetZCashUtxo(3);
+    auto result = PickZCashTransparentInputs(utxo_map, 10,
+                                             ZCashTargetOutputType::kOrchard);
+    EXPECT_FALSE(result);
+  }
```

Comment:
Can we add a test for when there's zero change returned to the user and the fee and the amount equal the input sum (e.g. no dust is left). Let's make sure that behavior matches what's expected on the network too